### PR TITLE
Allow configuring the Prow URL for command help

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Not all Prow deployments will use the configurations that prow.k8s.io
does, and it is best to link to the direct help page for the repo in
question for the deployment that is used to run it.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @spiffxp @cblecker @nikhita

### CODE REVIEW

File: prow/plugins/respond.go [hunk 1]
```diff
@@ -26,11 +26,13 @@ import (
 // AboutThisBotWithoutCommands contains the message that explains how to interact with the bot.
 const AboutThisBotWithoutCommands = "Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository."
```
Comment (stevekuznetsov) [inline prow/plugins/respond.go]: FYI I noticed all of the helpers in this file use this one, without the command help. Why? Do we want to standardize on using the full footer?

**Comment by cblecker** [comment]
It doesn't look like prow.k8s.io's config has the value of this set. Also, we should have a default value fallback, so that other consumers of prow can default to the current functionality, but with the ability to set their own URL.

**Comment by stevekuznetsov** [comment]
Not super thrilled about having `go.k8s.io` as the fallback since it's wrong in most cases, but ok.

**Comment by cblecker** [comment]
The other option would be to just link directly to `${deck_url}/command-help` if deck is installed?

**Comment by stevekuznetsov** [comment]
I don't think we have any way to know where `deck` is, without a configuration option like this. See also every config field in the main config struct that references the `deck` URL

**Comment by stevekuznetsov** [comment]
@spiffxp @nikhita any thoughts?

**Comment by stevekuznetsov** [comment]
OK, this got super messy by adding it to the controllers for reporting. @cjwagner what do you think we should be doing here?

**Comment by stevekuznetsov** [comment]
@munnerz i had to futz with your DCO tests a lot here since you were asserting equality on mangled comments missing org/repo :)

File: prow/plugins/approve/approve_test.go [hunk 2]
```diff
@@ -1128,10 +1129,9 @@ Approvers can cancel approval by writing ` + "`/approve cancel`" + ` in a commen
 				)
 			} else if expect, got := fmt.Sprintf("org/repo#%v:", prNumber)+test.expectedComment, fghc.IssueCommentsAdded[0]; test.expectedComment != "" && got != expect {
 				t.Errorf(
-					"[%s] Expected the created notification to be:\n%s\n\nbut got:\n%s\n\n",
+					"[%s] Expected a different created notification: %v",
 					test.name,
-					expect,
-					got,
+					diff.StringDiff(expect, got),
```
Comment (stevekuznetsov) [inline prow/plugins/approve/approve_test.go]: Type of error we used to get:

```
        dco_test.go:901: expected comment to be "org/repo#3:Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.\n\n:memo: **Please follow instructions in the [contributing guide](https://github.com///blob/master/CONTRIBUTING.md) to update your commits with the DCO**\n\nFull details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).\n\n**The list of commits missing DCO signoff**:\n\n- [sha](https://github.com///commits/sha) not a sign off\n\n<details>\n\nInstructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](http.com/bot-commands?repo=org%2Frepo).\n</details>\n" but it was "org/repo#3:Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.\n\n:memo: **Please follow instructions in the [contributing guide](https://github.com/org/repo/blob/master/CONTRIBUTING.md) to update your commits with the DCO**\n\nFull details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).\n\n**The list of commits missing DCO signoff**:\n\n- [sha](https://github.com/org/repo/commits/sha) not a sign off\n\n<details>\n\nInstructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](http.com/bot-commands?repo=org%2Frepo).\n</details>\n"
```

(unhelpful :) )

**Comment by stevekuznetsov** [comment]
@cblecker @spiffxp @nikhita any thoughts?

File: prow/external-plugins/refresh/main.go [hunk 3]
```diff
@@ -54,8 +52,8 @@ func (o *options) Validate() error {
 		}
 	}
 
-	if _, err := url.ParseRequestURI(o.prowURL); err != nil {
-		return fmt.Errorf("invalid -prow-url URI: %q", o.prowURL)
+	if len(o.prowURL) > 0 {
+		logrus.Warn("--prow-url is deprecated, set prow_url in config.yaml instead")
```
Comment (cblecker) [inline prow/external-plugins/refresh/main.go]: Do we need to go through the deprecation policy for this?

File: prow/plugins/config.go [hunk 4]
```diff
@@ -51,6 +51,9 @@ type Configuration struct {
 	// Owners contains configuration related to handling OWNERS files.
 	Owners Owners `json:"owners,omitempty"`
 
+	// ProwURL is the URL to use in help messages when referring to this deployment
+	ProwURL string `json:"prow_url"`
```
Comment (cblecker) [inline prow/plugins/config.go]: Do we need to have two separate settings for this? (one in config.yaml, and another in plugins.yaml?)

Comment (stevekuznetsov) [reply prow/plugins/config.go]: Not everyone loads both :(

It's awful, suggestions appreciated here (@cjwagner ?)

File: prow/external-plugins/refresh/main.go (hunk 3 above)
Comment (stevekuznetsov) [reply prow/external-plugins/refresh/main.go]: I doubt anyone but me uses `refresh` but yes, we can default here.

File: prow/plugins/config.go (hunk 4 above)
Comment (cjwagner) [reply prow/plugins/config.go]: That belongs in Prow config only (not plugin config) IMO. Core Prow config is also exposed to plugins and the Prow URL makes more sense in core config anyways.

**Pull request closed without merging**